### PR TITLE
Recursive Verifier

### PR DESCRIPTION
**Pull request by zhiyong1997**

### CODE REVIEW

File: gkr/src/prover/gkr.rs [hunk 1]
```diff
@@ -6,14 +6,14 @@ use circuit::Circuit;
 use config::{GKRConfig, MPIConfig};
 use polynomials::MultiLinearPoly;
 use sumcheck::{sumcheck_prove_gkr_layer, ProverScratchPad};
-use transcript::{Transcript, TranscriptInstance};
+use transcript::Transcript;
 
 // FIXME
 #[allow(clippy::type_complexity)]
-pub fn gkr_prove<C: GKRConfig>(
+pub fn gkr_prove<C: GKRConfig, T: Transcript<C::ChallengeField>>(
```
Comment (zhenfeizhang) [inline gkr/src/prover/gkr.rs:13]: I was thinking to have a single config file for GKR and Transcript will be part of this configuration.

File: recursive/main.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
+package main
```
Comment (zhenfeizhang) [inline recursive/main.go]: nits: can we use `recursive_circuit` or `recursion` as the name of this folder?

File: transcript/src/fiat_shamir_hash/mimc.rs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    let mut h = [0u8; 32];
+    keccak.update(seed.as_bytes());
+    keccak.finalize(&mut h);
+
+    for _ in 0..n_rounds {
+        let mut keccak = Keccak::v256();
+        keccak.update(&h);
+        keccak.finalize(&mut h);
+
+        // big endian -> little endian, in order to match the one in gnark
```
Comment (zhenfeizhang) [inline transcript/src/fiat_shamir_hash/mimc.rs:83]: we should have a test with same test vectors for gnark and expander

File: recursive/main.go (hunk 2 above)
Comment (zhiyong1997) [reply recursive/main.go]: Edited, will be in next PR.

File: transcript/src/fiat_shamir_hash/mimc.rs (hunk 3 above)
Comment (zhiyong1997) [reply transcript/src/fiat_shamir_hash/mimc.rs:83]: Added, will be in next pr.

**Review by zhenfeizhang — APPROVED** [review]
LGTM. Great work!

**Pull request merged**